src/Common/ThreadPool.cpp

```diff
@@ -272,22 +278,40 @@ void ThreadPoolImpl<Thread>::worker(typename std::list<Thread>::iterator thread_
 
         if (!need_shutdown)
         {
+            // set up tracing context for this thread by its parent context
```

Comment:
Why isn't that in the `try` block below (l. 285)? The ctor of `OpenTelemetryThreadTraceContextScope` is rather complex and can throw exceptions.

Reply:
The ctor looks like a little complex, but all are fields assignment and simple function call, I think they don't throw exception.

If we move it into the `try` block, we have to allocate it on heap.